pallets/loans/src/lib.rs

```diff
@@ -781,13 +774,61 @@ pub mod pallet {
 
 			Ok(Some(T::WeightInfo::update_portfolio_valuation(count)).into())
 		}
+
+		/// Transfer debt from one loan to another loan,
+		/// repaying from the first loan and borrowing the same amount from the
+		/// second loan. `from_loan_id` is the loan used to repay.
+		/// `to_loan_id` is the loan used to borrow.
+		/// The repaid and borrow amount must match.
+		#[pallet::weight(10_000_000)] //TODO
+		#[pallet::call_index(11)]
+		pub fn transfer_debt(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			from_loan_id: T::LoanId,
+			to_loan_id: T::LoanId,
+			repaid_amount: RepaidPricingAmount<T>,
+			borrow_amount: PricingAmount<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			ensure!(
+				from_loan_id != to_loan_id,
+				Error::<T>::TransferDebtToSameLoan
+			);
+
+			let repaid_amount = Self::repay_action(&who, pool_id, from_loan_id, &repaid_amount)?.0;
+
+			match &borrow_amount {
+				PricingAmount::Internal(value) => {
+					ensure!(
+						*value == repaid_amount.repaid_amount()?.total()?,
+						Error::<T>::TransferDebtAmountMismatched
+					)
+				}
+				PricingAmount::External(_external) => {
+					// TODO (1): handle it once slippage is added.
+					// TODO (2): if quantity is measured as a rate,
+					// then instead of using slippage we can use quantity with
+					// decimals.
+				}
+			}
```

Comment:
If we have the implicit invariant that says that repayment and borrowing ideally should withdraw/deposit the same exact amount from the pool, then it works fine for Internal pricing but would only work fine for External pricing if we chose case 2: Decimals with quantity. 

In that case, we should add that feature before merging this not to open a security hole allowing the transfer of debt amounts that are not equivalent to doing the extrinsics separately: `repay() + borrow()`.

Reply:
i.e. 
borrower does:
- repay with (`quantity * settlement_price_1`)
- borrow with (`quantity * settlement_price_2`)

Then debt has been transferred but they have more balance in their account if `settlement_price_2 > settlement_price_1`.

Instead, if the borrower does:
- transfer debt (repay: `quantity * settlement_price_1`, borrow: `quantity * settlement_price_2`)

Then debt has been transferred but their account balance remains the same when not.

Reply:
cc @offerijns @denniswell

Reply:
Solved once we use decimals for quantities